01_fib/hsy3418/main_test.go

```diff
@@ -0,0 +1,21 @@
+package main
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+
+	expected := strconv.Quote("1\n1\n2\n3\n5\n8\n13\n")
```

Comment:
Better to use multi-line string as follows. This will help improving readability.

```
strconv.Quote(`1
1
2
3
5
8
13
`)
```

Reply:
Thanks Kasun, but after i used the multiline string, the expected result becomes
 expected:"\"1\\n\\t1\\n\\t2\\n\\t3\\n\\t5\\n\\t8\\n\\t13\\n\\t\""
but my actual result is
actual:  "\"1\\n1\\n2\\n3\\n5\\n8\\n13\\n\""
How am I get rid of \t from the multiline string?

Reply:
You should not use indentation in multiline string. `\t` is tab character.

```
func TestMainOutput(t *testing.T) {
	var buf bytes.Buffer
	out = &buf

	main()

	expected := strconv.Quote(`1
1
2
3
5
8
13
`)
	actual := strconv.Quote(buf.String())

	if expected != actual {
		t.Errorf("Unexpected output in main()\nexpected: %q\nactual: %q", expected, actual)
	}
}
```